src/plugins/intel_gpu/include/intel_gpu/primitives/eye.hpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+#include "primitive.hpp"
+
+namespace cldnn {
+/// @addtogroup cpp_api C++ API
+/// @{
+/// @addtogroup cpp_topology Network Topology
+/// @{
+/// @addtogroup cpp_primitives Primitives
+/// @{
+
+/// @brief
+/// @details Construct identity matrix or batch fo them
+struct eye : public primitive_base<eye> {
+    CLDNN_DECLARE_PRIMITIVE(eye)
+
+    /// @brief Constructs eye primitive.
+    /// @param id This primitive id.
+    /// @param inputs List of primitive ids.
+    /// @param output_shape Tensor output shape
+    eye(const primitive_id& id,
+        const std::vector<primitive_id>& inputs,
+        const tensor& output_shape,
+        const primitive_id& ext_prim_id = "",
+        const padding& output_padding = padding())
+        : primitive_base{id, inputs, ext_prim_id, output_padding},
```

Comment:
You can pass "output_data_type" param in eye constructor and set it in "primitive_base" as well, so we will be sure that we have it set. Because you use it without checking whether it's set or not, so we can avoid crash if somebody forgot to set it.

Reply:
done